uc-crux-llvm: Panic on `fprintf`

```c
#include <stdio.h>

int main() {
  fprintf(stderr, "stderr!\n");
  return 0;
}
```
```llvm
; ModuleID = 'test.bc'
source_filename = "test.c"
target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"

%struct._IO_FILE = type { i32, i8*, i8*, i8*, i8*, i8*, i8*, i8*, i8*, i8*, i8*, i8*, %struct._IO_marker*, %struct._IO_FILE*, i32, i32, i64, i16, i8, [1 x i8], i8*, i64, %struct._IO_codecvt*, %struct._IO_wide_data*, %struct._IO_FILE*, i8*, i64, i32, [20 x i8] }
%struct._IO_marker = type opaque
%struct._IO_codecvt = type opaque
%struct._IO_wide_data = type opaque

@stderr = external local_unnamed_addr global %struct._IO_FILE*, align 8
@.str = private unnamed_addr constant [9 x i8] c"stderr!\0A\00", align 1

; Function Attrs: nounwind sspstrong uwtable
define i32 @main() local_unnamed_addr #0 {
  %1 = load %struct._IO_FILE*, %struct._IO_FILE** @stderr, align 8, !tbaa !3
  %2 = tail call i32 (%struct._IO_FILE*, i32, i8*, ...) @__fprintf_chk(%struct._IO_FILE* %1, i32 1, i8* getelementptr inbounds ([9 x i8], [9 x i8]* @.str, i64 0, i64 0)) #2
  ret i32 0
}

declare i32 @__fprintf_chk(%struct._IO_FILE*, i32, i8*, ...) local_unnamed_addr #1
```
yields
```
You have encountered a bug in Crucible's implementation.
*** Please create an issue at https://github.com/GaloisInc/crucible/issues

%< ---------------------------------------------------
  Revision:  375ba043ccbc451fe116fb45c894987588ae80f7
  Branch:    lb/override-ty-panic (uncommited files present)
  Location:  Intrinsics.transformLLVMRet
  Message:   Cannot transform types
             *** Source type: VectorRepr AnyRepr
             *** Target type: IntrinsicRepr LLVM_pointer [BVRepr 64]
             in function: __fprintf_chk
CallStack (from HasCallStack):
  panic, called at src/Lang/Crucible/Panic.hs:11:9 in crucible-0.6-inplace:Lang.Crucible.Panic
  panic, called at src/Lang/Crucible/LLVM/Intrinsics/Common.hs:191:5 in crucible-llvm-0.4-inplace:Lang.Crucible.LLVM.Intrinsics.Common
%< ---------------------------------------------------
```

What version of Clang are you using? I get an ordinary error message with Clang 10:

```
$ clang --version
clang version 10.0.0-4ubuntu1 
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```
```
$ ~/Software/crux-llvm-0.6/bin/crux-llvm test.c
[Crux] Using pointer width: 64 for file crux-build/crux~test.bc
[Crux] Simulating function main
[Crux] Attempting to prove verification conditions.
[Crux] *** debug executable: results/test/debug-4
[Crux] *** break on line: 4
[Crux] Found counterexample for verification goal
[Crux]   test.c:4:11: error: in main
[Crux]   Error during memory load
[Crux]     No previous write to this location was found
[Crux]       Attempting load at type: i64
[Crux]     Performing overall load at type: i64
[Crux]       Via pointer: Global symbol "stderr" (3, 0x0:[64])
[Crux]     In memory state:
[Crux]       Stack frame main
[Crux]         No writes or allocations
[Crux]       Base memory
[Crux]         Allocations:
[Crux]           GlobalAlloc 4 0x9:[64] Immutable 1-byte-aligned [global variable  ] .str
[Crux]           GlobalAlloc 3 0x8:[64] Mutable 8-byte-aligned [global variable  ] stderr
[Crux]           GlobalAlloc 2 0x0:[64] Immutable 1-byte-aligned [defined function ] main
[Crux]           GlobalAlloc 1 0x0:[64] Immutable 1-byte-aligned [external function] fwrite
[Crux]         Writes:
[Crux]           Indexed chunk:
[Crux]             4 |->   *(4, 0x0:[64]) := "stderr!\n\NUL"
[Crux]     in context:
[Crux]       main
[Crux] Goal status:
[Crux]   Total: 1
[Crux]   Proved: 0
[Crux]   Disproved: 1
[Crux]   Incomplete: 0
[Crux]   Unknown: 0
[Crux] Overall status: Invalid.
```

--- a/uc-crux-llvm/src/UCCrux/LLVM/Overrides/Skip.hs
+++ b/uc-crux-llvm/src/UCCrux/LLVM/Overrides/Skip.hs
@@ -50,6 +50,7 @@ import           Lang.Crucible.FunctionHandle (SomeHandle(..), handleMapToHandle
 import           Lang.Crucible.Simulator.ExecutionTree (functionBindings, stateContext, fnBindings)
 import qualified Lang.Crucible.Simulator as Crucible
 import qualified Lang.Crucible.Simulator.OverrideSim as Override
+import qualified Lang.Crucible.Types as CTy
 
 -- crucible-llvm
 import           Lang.Crucible.LLVM.Extension (LLVM)
@@ -71,6 +72,7 @@ import           UCCrux.LLVM.ExprTracker (ExprTracker)
 import           UCCrux.LLVM.FullType.FuncSig (FuncSigRepr(FuncSigRepr))
 import qualified UCCrux.LLVM.FullType.FuncSig as FS
 import           UCCrux.LLVM.FullType.Type (MapToCrucibleType)
+import qualified UCCrux.LLVM.FullType.VarArgs as VA
 import           UCCrux.LLVM.Module (FuncSymbol, funcSymbol, makeFuncSymbol, isDebug, funcSymbolToString)
 import           UCCrux.LLVM.Overrides.Polymorphic (PolymorphicLLVMOverride, makePolymorphicLLVMOverride)
 import           UCCrux.LLVM.Postcondition.Apply (applyPostcond)
@@ -137,6 +139,7 @@ unsoundSkipOverrides modCtx bak mtrans usedRef annotationRef postconds decls =
                           (Post.minimalUPostcond ret)
                           (Map.lookup funcSym postconds)
               in createSkipOverride modCtx bak usedRef annotationRef post funcSym
+
     pure $
       map
         create
@@ -164,8 +167,24 @@ mkOverride ::
     IO (Crucible.RegValue sym (FS.ReturnTypeToCrucibleType arch mft), MemImpl sym)) ->
   PolymorphicLLVMOverride arch (personality sym) sym
 mkOverride modCtx _proxy funcSymb impl =
+  -- There's a lot of duplication here because the case over whether or not the
+  -- function is varargs can't be pushed further down; doing so causes "type is
+  -- untouchable inside..." errors
   case modCtx ^. funcTypes . funcSymbol funcSymb of
-    Some fs@(FuncSigRepr _ argFTys retTy) ->
+    Some fs@(FuncSigRepr VA.IsVarArgsRepr argFTys retTy) ->
+      case assignmentToCrucibleType modCtx argFTys of
+        SomeAssign' argTys Refl _ ->
+          makePolymorphicLLVMOverride $
+            basic_llvm_override $
+              LLVMOverride
+                { llvmOverride_declare = decl,
+                  llvmOverride_args = argTys Ctx.:> CTy.VectorRepr CTy.AnyRepr,
+                  llvmOverride_ret = toCrucibleReturnType modCtx retTy,
+                  llvmOverride_def =
+                    \mvar _sym (args Ctx.:> _) ->
+                      Override.modifyGlobal mvar (\mem -> liftIO (impl fs mem args))
+                }
+    Some fs@(FuncSigRepr VA.NotVarArgsRepr argFTys retTy) ->
       case assignmentToCrucibleType modCtx argFTys of
         SomeAssign' argTys Refl _ ->
           makePolymorphicLLVMOverride $

